core/src/main/scala/org/scalarelational/model/SQLDatastore.scala

```diff
@@ -90,7 +90,7 @@ abstract class SQLDatastore protected() extends Datastore {
     b.append(column.name)
     b.append(' ')
     b.append(column.sqlType)
-    if (column.has(NotNull)) {
+    if (!column.isOptional && !column.has(Polymorphic)) {
       b.append(" NOT NULL")
```

Comment:
Just at a glance, but I believe the null will come from the value as I don't see any exclusion of the value.